From 02f21e1cf6b1ddd7802f568cdc7b0d84ca82e747 Mon Sep 17 00:00:00 2001
From: Zhou Furong <furong.zhou@intel.com>
Date: Mon, 26 Nov 2018 13:26:16 +0800
Subject: [PATCH 0863/1214] Revert "cbc: Avoid rx sequence counter mismatch
 warnings"

This reverts commit b43dcb4264fc98ee3ecf5c900f192bce0a1edac9.
Signed-off-by: Zhou Furong <furong.zhou@intel.com>
Tracked-On: PKT-1557
---
 drivers/tty/cbc/cbc_link_layer.c | 30 ++++++++++++++++--------------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/drivers/tty/cbc/cbc_link_layer.c b/drivers/tty/cbc/cbc_link_layer.c
index bd747aa..a55cd8c 100644
--- a/drivers/tty/cbc/cbc_link_layer.c
+++ b/drivers/tty/cbc/cbc_link_layer.c
@@ -264,8 +264,7 @@ u8 cbc_core_on_receive_cbc_serial_data(u8 length, const u8 *rx_buf)
 	return number_of_bytes_accepted;
 }
 
-static void _cbc_link_layer_checksum(u8 *rx_cvh_frame, u8 frame_length,
-						struct cbc_buffer *buffer)
+static void _cbc_link_layer_checksum(u8 *rx_cvh_frame, u8 frame_length)
 {
 	u8 expected_checksum = 0U;
 	u8 checksum = 0U;
@@ -303,17 +302,6 @@ static void _cbc_link_layer_checksum(u8 *rx_cvh_frame, u8 frame_length,
 			rx_cvh_frame[1]
 			& CBC_SEQUENCE_COUNTER_WIDTH_MASK;
 		}
-
-		/* Increment seq. counter. */
-		rx_sequence_counter++;
-		rx_sequence_counter &=
-		CBC_SEQUENCE_COUNTER_WIDTH_MASK;
-
-		/* Forward frame to Mux. layer. */
-		buffer->frame_length = frame_length;
-		cbc_mux_multiplexer_process_rx_buffer(buffer);
-		cbc_link_release_rx_data(frame_length);
-		last_rx_frame_valid = 1;
 	}
 
 }
@@ -373,7 +361,21 @@ void cbc_link_layer_rx_handler(void)
 			} else if (bytes_avail >= frame_length) {
 				/* ok */
 				_cbc_link_layer_checksum(rx_cvh_frame,
-							 frame_length, buffer);
+							 frame_length);
+
+				/* Increment seq. counter. */
+				rx_sequence_counter++;
+				rx_sequence_counter &=
+				CBC_SEQUENCE_COUNTER_WIDTH_MASK;
+
+				/* Forward frame to Mux. layer. */
+				buffer->frame_length = frame_length;
+				cbc_mux_multiplexer_process_rx_buffer(
+							buffer);
+				cbc_link_release_rx_data(frame_length);
+
+					last_rx_frame_valid = 1;
+				 /* else */
 				number_of_bytes_expected = 0;
 			} else {
 				/*
-- 
2.7.4

